components/Profile/SettingsModal.jsx

```diff
@@ -78,12 +109,47 @@ const SettingsModal = (props) => {
       >
 
         <CenterView onPress={() => setModalVisible(!modalVisible)}>
-          <ModalView onPress={() => setModalVisible(modalVisible)}>
+          <ModalView onPress={() => {
+            setModalVisible(modalVisible);
+            Keyboard.dismiss();
+          }}
+          >
             <Pressable onPress={() => setModalVisible(false)}>
               <Ionicons name="ios-close" size={triggerSize} color="black" />
             </Pressable>
             <Title>Settings</Title>
-            <SubTitle>Notifications</SubTitle>
+
+            <SubTitle>Rest Notifications</SubTitle>
+            <TwinView>
+              <SettingLabel>Enable Notifications:</SettingLabel>
+              <Switch
+                onValueChange={() => {
+                  setSettingState({
+                    ...settingState,
+                    enableRestNotifications: !settingState.enableRestNotifications,
+                  });
+                }}
+                value={settingState.enableRestNotifications}
+              />
+            </TwinView>
+
+            <TwinView>
+              <SettingLabel disabled={!settingState.enableRestNotifications}>Rest Notification Timer:</SettingLabel>
```

Comment:
Also, if you enable notifications, then clear the input field and save changes -- it'll show as NaN

Reply:
I'm pretty sure this also causes a "Malformed calls from JS: field sizes are different" error when you click done on a set.